addons/purchase_stock/models/purchase.py

```diff
@@ -95,7 +95,7 @@ def write(self, vals):
 
     def button_approve(self, force=False):
         result = super(PurchaseOrder, self).button_approve(force=force)
-        self._create_picking()
+        self.filtered(lambda po: po.state in ('purchase', 'done'))._create_picking()
```

Comment:
I would rather move this filter into _create_picking directly in case this method is reused elsewhere

Reply:
Done